ffi/ffi.h

```diff
@@ -0,0 +1,101 @@
+#ifndef __LIBFF_FFI_FFI_H__
+#define __LIBFF_FFI_FFI_H__
+
+#include <stddef.h>
+
+// Externally visible entry points for FFI implementations.
+
+#if __cplusplus
+extern "C"
+{
+#endif
```

Comment:
Do you need to keep the `extern "C"` directives in the `cpp` file above if you already use an `extern "C"` block for c++ header code here (I think this is enough to indicate it here for the linker, no?)?

Reply:
Yes, it should be enough.  Once it has been declared as `extern "C"` the compiler should obey that for the implementation, I believe.  (Even if there was a name-mangled version in the global namespace, the language provides no way to choose which function to call)

Reply:
See https://github.com/clearmatics/libff/pull/33#discussion_r546658523